### PR TITLE
feat(unoin schema): Genereate union MAE schema

### PR DESCRIPTION
**Pull request by zhixuanjia**
## TLDR
We already have support for union MCE. Similar idea can be extended to MAE. This PR add capability to generate union MAE schema.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Review by yangyangv2 — APPROVED** [review]
LGTM.

Open Question: should we consider deprecating FailedMCE? given all the failures are going to the DLQ?
